pkg/operatorsource/validating_phase_reconciler.go

```diff
@@ -0,0 +1,31 @@
+package operatorsource
+
+import (
+	"context"
+
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+)
+
+// NewValidatingPhaseReconciler returns a PhaseReconciler that reconciles
+// an OperatorSource object in "Validating" phase
+func NewValidatingPhaseReconciler() PhaseReconciler {
+	return &validatingPhaseReconciler{}
+}
+
+type validatingPhaseReconciler struct{}
+
+// Reconcile reconciles an OperatorSource object that is in "Validating" phase.
+// It ensures that the given object is valid before it is scheduled for download.
```

Comment:
How does it ensure that a given object is valid? I am unable to understand why we need this phase the way things stand today. Is it something you feel might be needed in the future? If that is the case please introduce this phase only we are really going to be using it.

Reply:
Today, we perform a simple check to make sure `Endpoint` is a valid URL.

Reply:
Sorry I missed that. I will take a closer look when reviewing.